### PR TITLE
haskell-backend: update submodule

### PR DESCRIPTION
**Pull request by ehildenb**

### CODE REVIEW

**Review by dwightguth — APPROVED** [review]

**Comment by ehildenb** [comment]
Confirmed that this works for passing Haskell-backend tests on KEVM.

**Comment by ehildenb** [comment]
Wait on merging until I get a fix for `generatedCounter` into this.

File: kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ public static String getKompiledString(Module mainModule, KLabel topCellInitiali
         DefinitionTransformer generateSortPredicateSyntax = DefinitionTransformer.from(new GenerateSortPredicateSyntax()::gen, "adding sort predicate productions");
         DefinitionTransformer subsortKItem = DefinitionTransformer.from(Kompile::subsortKItem, "subsort all sorts to KItem");
         DefinitionTransformer expandMacros = DefinitionTransformer.fromSentenceTransformer((m, s) -> new ExpandMacros(m, files, kompileOptions, false).expand(s), "expand macros");
-        Function1<Definition, Definition> resolveFreshConstants = d -> DefinitionTransformer.from(new ResolveFreshConstants(d, true)::resolve, "resolving !Var variables").apply(d);
+        Function1<Definition, Definition> resolveFreshConstants = d -> DefinitionTransformer.from(m -> GeneratedTopFormat.resolve(new ResolveFreshConstants(d, true).resolve(m)), "resolving !Var variables").apply(d);
         DefinitionTransformer generatedTopFormat = DefinitionTransformer.from(GeneratedTopFormat::resolve, "setting generatedTop format attribute");
```
Comment (dwightguth) [inline kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java]: shouldn't you remove this line?

Comment (ehildenb) [reply kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java]: Good catch, fixed in both Kompile and KoreBackend.

**Review by dwightguth — APPROVED** [review]

**Comment by ehildenb** [comment]
Conformance tests passing on KEVM (using haskell backend): https://office.runtimeverification.com/jenkins/job/evm-semantics/job/PR-312/

**Pull request merged**